### DISCUSSION TITLE
[ Aptos CLI for type argument ]

### DISCUSSION THREAD

**Discussion by ctb0k33** [aptos-labs/aptos-developer-discussions#222] [Questions] [answered]
### Discord user ID

ctb0k33

### Describe your question in detail.

I currently develop a minting smart contract that support multiple coinType.
Here is my helper function in my smart contract that use to change coinType and mintPrice
```move
  public fun update_state<coinType>(signer: &signer, new_mint_price:u64) acquires State {
        // Get the object that represents the module
        let module_object = object::address_to_object<ObjectCore>(@my_module);

        // Only owner can change the state of the module
        assert!(object::owner(module_object)==signer::address_of(signer),0);
        let state = borrow_global_mut<State>(@my_module);
        state.mintPrice = new_mint_price;
        state.coin = type_of<coinType>()
    }
 ```
 
 And the script the wrote to call the update_state function
 ```move
 script{
    use my_module::my_module;
     fun update_state<coinType>(sender: &signer,new_mint_price:u64){
         my_module::update_state<coinType>(sender,new_mint_price);
     }
}
```

The CLI:
aptos move run-script --assume-yes --profile default --compiled-script-path build/my_module/bytecode_scripts/update_state.mv --args u64:1000000 --type-args 0xcf8237d41d703da53d7772d882146ff7b49d237d8a96514d0fb842c12d4b207d::moon_coin::MoonCoin

And i got the error: "  "Error": "Simulation failed with status: TYPE_MISMATCH"
All of the contract are deployed in devnet

### What error, if any, are you getting?

As describe above

### What have you tried or looked at?  Or how can we reproduce the error?

As describe above

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xjunha** [comment]
Can you provide link to your deployed smart contract?

**Reply by ctb0k33** [reply]
https://aptoscan.com/account/0xcf8237d41d703da53d7772d882146ff7b49d237d8a96514d0fb842c12d4b207d?viewMode=view&module=moon_coin&network=devnet#modules

Here is the link to moon_coin module @0xjunha

**Reply by 0xjunha** [reply]
The link to the module where `update_state` function is implemented?

**Reply by ctb0k33** [reply]
https://aptoscan.com/objects/0xcc93ae03b09a1a4e11b2d9497b83e4caf47bca4650c89005b4d5569f660a1e29?network=devnet

Here is the simplify of the contract

Module address: https://aptoscan.com/module/0xcc93ae03b09a1a4e11b2d9497b83e4caf47bca4650c89005b4d5569f660a1e29/mint?network=devnet

**Comment by ctb0k33** [comment]
Due to devnet release, here is the new link to the contract:
https://aptoscan.com/module/0xddc866b708c40a7c0f0e7f84badc592890c99c8ea7cc2be91b6d3de37d99b06e/mint?network=devnet
@0xjunha please help !

**Reply by 0xjunha** [reply, marked as answer]
I tried to reproduce your error but couldn't find that error message. The script runs successfully on my end. Could you check if all your module address aliases (those used for deploying mint.move and running update_state.move) are provided correctly? In other words, please ensure your script is calling functions from the `mint.move` module as expected.

CLI command I used:
`aptos move run-script --assume-yes --profile default --compiled-script-path build/mint/bytecode_scripts/update_state.mv --args u64:1000000 --type-args 0x79c9d74fa766ed7bb29231196afe46fd9ea1de29fb6f6d5e0a5fb78921969f7e::moon_coin::MoonCoin`


Here're links to my tx run:

- [Object where mint.move module is deployed](https://aptoscan.com/module/0xa6325e4cfbda26e579edcc882d0700ebaf5c92a72ae67d759d42087b6256c467/mint?network=devnet)

- [MoonCoin](https://aptoscan.com/module/0x79c9d74fa766ed7bb29231196afe46fd9ea1de29fb6f6d5e0a5fb78921969f7e/moon_coin?network=devnet)

- [update_state script run tx](https://aptoscan.com/transaction/5231038?network=devnet#payload)

**Reply by ctb0k33** [reply]
@0xjunha  It's work now. But can i transfer the type of TypeInfo to CoinType or sth like that so that i can directly pass the coin variable in State struct to transfer function without using the generic version of mint function. Thanks !

**Reply by ctb0k33** [reply]
Already found the solution for my problem. Thanks for help !
